overlord/devicestate/devicemgr_test.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package devicestate_test
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/assertstest"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/release"
+)
+
+func TestDeviceManager(t *testing.T) { TestingT(t) }
+
+type deviceMgrSuite struct {
+	state *state.State
+	mgr   *devicestate.DeviceManager
+
+	storeSigning *assertstest.StoreStack
+}
+
+var _ = Suite(&deviceMgrSuite{})
+
+func (s *deviceMgrSuite) SetUpTest(c *C) {
+	dirs.SetRootDir(c.MkDir())
+
+	rootPrivKey, _ := assertstest.GenerateKey(1024)
+	storePrivKey, _ := assertstest.GenerateKey(752)
+	s.storeSigning = assertstest.NewStoreStack("canonical", rootPrivKey, storePrivKey)
+
+	s.state = state.New(nil)
+	mgr, err := devicestate.Manager(s.state)
+	c.Assert(err, IsNil)
+
+	s.mgr = mgr
+}
+
+func (s *deviceMgrSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("")
+}
+
+func (s *deviceMgrSuite) settle() {
+	for i := 0; i < 50; i++ {
+		s.mgr.Ensure()
+		s.mgr.Wait()
+	}
+}
+
+func (s *deviceMgrSuite) TestFullDeviceRegistrationHappyOnClassic(c *C) {
```

Comment:
Test feels a bit short considering how much logic there is above, but again perhaps this says more about the fact there's more logic than necessary than about the test being improper.